src/client.ts

```diff
@@ -17,25 +17,17 @@ import { addUrlQuery, defaultPortalUrl, makeUrl } from "./utils";
 import { getFile, setFile } from "./skydb";
 import { lookupRegistry, updateRegistry } from "./registry";
 
+export type CustomClientOptions = {
+  APIKey?: string; // authentication password to use
+  customUserAgent?: string; // custom user agent header to set
+  onUploadProgress?: (progress: number, event: ProgressEvent) => void; // optional callback to track upload progress
+};
+
 export class SkynetClient {
   portalUrl: string;
-  customOptions: Record<string, unknown>;
+  customOptions: CustomClientOptions;
 
-  /**
-   * The Skynet Client which can be used to access Skynet.
-   * @constructor
-   * @param {string} [portalUrl="https://siasky.net"] - The portal URL to use to access Skynet, if specified. To use the default portal while passing custom options, use "".
```

Comment:
Why remove all this documentation instead of just the irrelevant parameters?

Reply:
yeah sorry, my bad

Reply:
Should we indicate that these parameters are optional? (e.g. with brackets in the JSDoc or a `?` after the parameter name)

Reply:
yeah, wrapped back in brackets